core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala

```diff
@@ -97,6 +99,10 @@ private[deploy] class Worker(
   private val APP_DATA_RETENTION_SECONDS =
     conf.getLong("spark.worker.cleanup.appDataTtl", 7 * 24 * 3600)
 
+  // Whether or not cleanup the non-shuffle files on executor death.
+  private val CLEANUP_NON_SHUFFLE_FILES_ENABLED =
+    conf.getBoolean("spark.storage.cleanupFilesAfterExecutorDeath", true)
```

Comment:
Shall we rename this config to "spark.storage.cleanupFilesAfterExecutorExit"? Seems from the code that normal executor exit (dynamic allocation) will also trigger the cleanup, this config may be a little misleading. Please correct me if I'm wrong.

Reply:
@jerryshao, that's a good point; I prefer your suggested `*Exit` name.

If executors exit in a healthy / controlled fashion then we don't expect non-shuffle files to be leaked, but the cleanup logic still will be run.